### PR TITLE
chore: update type drawer

### PR DESCRIPTION
**Pull request by jawn-ha**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: types/drawer.d.ts [hunk 1]
```diff
@@ -2,118 +2,123 @@
...
+     * @type string
+     */
+    placement?: 'top' | 'right' | 'bottom' | 'left';
+    afterVisibleChange?: (visible: boolean) => void;
+    keyboard?: boolean;
+
+    /**
+     * Specify a callback that will be called when a user clicks mask, close button or Cancel button.
+     */
+    close?: (e: Event) => void;
```
Comment (Amour1688) [inline types/drawer.d.ts:122]: `onClose?: (e?: Event) => void;

File: types/drawer.d.ts [hunk 2]
```diff
@@ -2,118 +2,123 @@
...
+     */
+    zIndex?: number;
+
+    /**
+     * The placement of the Drawer.
+     * @default 'right'
+     * @type string
+     */
+    placement?: 'top' | 'right' | 'bottom' | 'left';
+    afterVisibleChange?: (visible: boolean) => void;
```
Comment (Amour1688) [inline types/drawer.d.ts:116]: `(visible?: boolean) => void`

Comment (jawn-ha) [reply types/drawer.d.ts]: fix in commit 18a207a

File: types/drawer.d.ts [hunk 3]
```diff
@@ -2,118 +2,123 @@
...
+     * @type string
+     */
+    placement?: 'top' | 'right' | 'bottom' | 'left';
+    afterVisibleChange?: (visible?: boolean) => void;
+    keyboard?: boolean;
+
+    /**
+     * Specify a callback that will be called when a user clicks mask, close button or Cancel button.
+     */
+    onClose?: (e: Event) => void;
```
Comment (Amour1688) [inline types/drawer.d.ts:122]: e?: Event

Comment (jawn-ha) [reply types/drawer.d.ts]: fix on 2f29845

File: types/drawer.d.ts [hunk 4]
```diff
@@ -2,118 +2,123 @@
...
+     * @type boolean
+     */
+    destroyOnClose?: boolean;
+
+    /**
+     * Return the mounted node for Drawer.
+     * @default 'body'
+     * @type any ( HTMLElement| () => HTMLElement | string)
+     */
+    getContainer?: () => HTMLElement | HTMLElement | string;
```
Comment (tangjinzhou) [inline types/drawer.d.ts:29]: duplicate HTMLElement

Comment (jawn-ha) [reply types/drawer.d.ts]: fux it on 9d14d05

**Pull request merged**